sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseService.java

```diff
@@ -880,62 +888,119 @@ public Predicate createDefaultFilter(CriteriaBuilder cb, From<?, Case> root) {
-			contact.setResultingCase(null);
-			externalJournalService.handleExternalJournalPersonUpdateAsync(contact.getPerson().toReference());
-			contactService.ensurePersisted(contact);
-		}
+		// Delete all tasks associated with this case
+		Optional.ofNullable(caze.getTasks()).ifPresent(tl -> tl.forEach(t -> taskService.deletePermanent(t)));
 
+		// Delete all samples that are only associated with this case
 		caze.getSamples()
 			.stream()
 			.filter(sample -> sample.getAssociatedContact() == null && sample.getAssociatedEventParticipant() == null)
-			.forEach(sample -> sampleService.delete(sample));
+			.forEach(sample -> sampleService.deletePermanent(sample));
 
-		// Delete all tasks associated with this case
-		List<Task> tasks = taskService.findBy(new TaskCriteria().caze(new CaseReferenceDto(caze.getUuid())), true);
-		for (Task task : tasks) {
-			taskService.deletePermanent(task);
-		}
+		// Delete surveillance reports related to this case
+		surveillanceReportService.getByCaseUuids(Collections.singletonList(caze.getUuid()))
+			.forEach(s -> surveillanceReportService.deletePermanent(s));
 
-		// Delete all prescriptions/treatments/clinical visits
+		// Delete documents related to this case
+		documentService.getRelatedToEntity(DocumentRelatedEntityType.CASE, caze.getUuid()).forEach(d -> documentService.markAsDeleted(d));
+
+		// Delete clinical management data
 		if (caze.getTherapy() != null) {
 			TherapyReferenceDto therapy = new TherapyReferenceDto(caze.getTherapy().getUuid());
-			treatmentService.findBy(new TreatmentCriteria().therapy(therapy)).stream().forEach(t -> treatmentService.deletePermanent(t));
-			prescriptionService.findBy(new PrescriptionCriteria().therapy(therapy)).stream().forEach(p -> prescriptionService.deletePermanent(p));
+			List<Treatment> treatments = treatmentService.findBy(new TreatmentCriteria().therapy(therapy));
+			treatments.forEach(t -> treatmentService.deletePermanent(t));
+			prescriptionService.findBy(new PrescriptionCriteria().therapy(therapy)).forEach(p -> prescriptionService.deletePermanent(p));
 		}
+
 		if (caze.getClinicalCourse() != null) {
 			ClinicalCourseReferenceDto clinicalCourse = new ClinicalCourseReferenceDto(caze.getClinicalCourse().getUuid());
-			clinicalVisitService.findBy(new ClinicalVisitCriteria().clinicalCourse(clinicalCourse))
-				.stream()
-				.forEach(c -> clinicalVisitService.deletePermanent(c));
+			List<ClinicalVisit> cvs = clinicalVisitService.findBy(new ClinicalVisitCriteria().clinicalCourse(clinicalCourse));
+			cvs.forEach(c -> clinicalVisitService.deletePermanent(c));
 		}
 
-		// Remove all events linked to case by removing the case_id from event participant
-		caze.getEventParticipants().stream().forEach(eventParticipant -> eventParticipant.setResultingCase(null));
+		// Remove the case from any S2S share info referencing it
+		sormasToSormasShareInfoService.getByAssociatedEntity(SormasToSormasShareInfo.CAZE, caze.getUuid()).forEach(s -> {
```

Comment:
As there is always one share info for one shared object, share info never has multiple shared objects linked,  it would be better to delete the share info. 

Doesn't make sense to keep share infos not linked to any shared object.

Also other systems should be notified that the case has been deleted so the will not try to sync back later changes.

Reply:
Unsubscribing from the source system is currently out of scope (#7736).